lib/IRGen/IRGenSIL.cpp

```diff
@@ -2053,38 +2053,87 @@ void IRGenSILFunction::visitGraphOperationInst(GraphOperationInst *i) {
     checkOk(status);
   };
 
-  // Handle inputs.
+  // Handle inputs and dynamic attributes.
   for (auto structuredArgument : opInfo.getStructuredArguments()) {
-    assert(structuredArgument.getArgumentNameWithSuffix().empty() &&
-           "cannot lower named arguments");
-    switch (structuredArgument.getKind()) {
-    case GraphOperationInfo::SAK_Single: {
-      LLVM_DEBUG(llvm::dbgs() << " Adding a single input.\n");
-      auto tensorHandle = structuredArgument.getSingleArgument();
-      addTensorhandleAsOpInput(tensorHandle);
+    auto argumentNameAndLowering =
+        structuredArgument.getArgumentNameAndLowering();
+    std::string argumentName = std::get<0>(argumentNameAndLowering);
+    auto argumentLowering = std::get<1>(argumentNameAndLowering);
+    LLVM_DEBUG(llvm::dbgs() << "IRGen graph_op argument " << argumentName
+               << "with lowering " << (unsigned)argumentLowering);
+    switch (argumentLowering) {
+    case GraphOperationInfo::ArgumentLowering::Out:
+      assert(0 && "Attributes cannot be output parameters");
+    case GraphOperationInfo::ArgumentLowering::UnknownShapeListAttribute:
+      assert(0 && "UnknownShapeListAttribute should have been eliminated "
+                  "by deabstraction");
+    case GraphOperationInfo::ArgumentLowering::TypeListAttribute:
+      assert(0 && "TypeListAttribute should have been eliminated by "
+                  "deabstraction");
+    case GraphOperationInfo::ArgumentLowering::TensorAttribute:
+      // See GraphOperationInfo.h for more information on why TensorAttribute
```

Comment:
Are you referring to the comment on the TensorAttribute enum decl? If so, can you make that explicit?

Reply:
I'll clarify the comment